centreon/src/Centreon/Infrastructure/MonitoringServer/MonitoringServerRepositoryRDB.php

```diff
@@ -269,6 +269,61 @@ public function findServer(int $monitoringServerId): ?MonitoringServer
         return null;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function findServerByIdAndAccessGroups(int $monitoringServerId, array $accessGroups): ?MonitoringServer
+    {
+        if ($accessGroups === []) {
+            return null;
+        }
+
+        $accessGroupIds = array_map(
+            fn($accessGroup) => $accessGroup->getId(),
+            $accessGroups
+        );
+
+        [$bindValues, $bindQuery] = $this->createMultipleBindQuery($accessGroupIds, ':acl_group_id_');
+
+        $request = $this->translateDbName(
+            <<<SQL
+                SELECT * FROM `:db`.nagios_server
```

Comment:
pollers ACLs are a filter, you need to check if the `acl_resources_poller_relations` is empty for accessGroups, then user has access to all pollers

Reply:
Done